src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -350,13 +351,20 @@ private SdkHttpClient getSdkHttpClient() {
      * @param thing         thing info
      * @param awsRegion     aws region
      * @param roleAliasName role alias for using IoT credentials endpoint
+     * @param certPath the path of certificates which users specify
      * @throws IOException                  Exception while reading root CA from file
      * @throws DeviceConfigurationException when the configuration parameters are not valid
      */
     public void updateKernelConfigWithIotConfiguration(Kernel kernel, ThingInfo thing, String awsRegion,
-                                                       String roleAliasName)
+                                                       String roleAliasName, String certPath)
             throws IOException, DeviceConfigurationException {
         Path rootDir = kernel.getNucleusPaths().rootPath();
+
+        if (!Utils.isEmpty(certPath)) {
+            rootDir = Paths.get(certPath);
+            Utils.createPaths(rootDir);
+        }
+
         Path caFilePath = rootDir.resolve("rootCA.pem");
         Path privKeyFilePath = rootDir.resolve("privKey.key");
         Path certFilePath = rootDir.resolve("thingCert.crt");
```

Comment:
nit: rename `rootDir` to `certPath` and rename `certPath` to maybe `userCertPath`, since it's no longer just root directory